examples/deepspeed/dcgan/model.py

```diff
@@ -46,7 +46,7 @@ def __init__(self, context: det_ds.DeepSpeedTrialContext,
         self.discriminator = self.context.wrap_model_engine(discriminator)
         self.fixed_noise = self.context.to_device(
             torch.randn(
-                self.context.train_micro_batch_size_per_gpu, self.hparams["noise_length"], 1, 1
```

Comment:
was this a bug, or is this a breaking change to the API?

Reply:
both? it was a bug because it was a breaking change. added to release notes.

Reply:
This is the only reason we are breaking existing DeepSpeedTrials?  Like, we're deprecating old paths, yes, but we're only actually breaking user code for this?

Can we include a `@property`-style getter that makes this _not_ a breaking change?

something like:

```python
    @property
    def train_micro_batch_size_per_gpu(self) -> int:
        warnings.warn(
            "DeepSpeedTrialContext.train_micro_batch_size_per_gpu has been deprecated in "
            "Determined 0.38.0; please use the context.get_train_micro_batch_size_per_gpu() getter "
            "instead."
            FutureWarning,
            stacklevel=2,
        )
        return self.get_train_micro_batch_size_per_gpu()
```